### ISSUE TITLE
Potential Data Leakage in data/data_loader.py due to Preprocessing Before Split

### ISSUE THREAD

**Issue by Sohrabbeig** [aikunyi/FourierGNN#10]
I've been reviewing the data preprocessing steps in `data/data_loader.py` and noticed that the entire dataset undergoes fitting and transformation before being split into training, validation, and test sets. This process might lead to data leakage, where information from the test and validation sets inadvertently influences the training process.

Is this approach an intentional part of the model's design for a specific reason that I might have missed? Or could it be an oversight?

**Comment by aikunyi** [comment]
we use the min/max value of training sets to normalize the validation/test sets

**Comment by yasinuygun** [comment]
When I look at the [code](https://github.com/aikunyi/FourierGNN/blob/882f23aaf77b5818b3ecc46bc58c5b96259f1164/data/data_loader.py#L72) at `data_loader.py`, the full data is being used to normalize:

<img width="563" alt="image" src="https://github.com/aikunyi/FourierGNN/assets/28067605/d43a97ff-22f4-4865-987e-f601cf8e2365">

I guess it should have been like the following instead:
```python
            training_end = int(len(data) * self.train_ratio)
            mms.fit(data[:training_end])
            data = mms.transform(data)
```

**Comment by aikunyi** [comment]
Thanks for the correction, we've revised it

**Comment by tolinlaws** [comment]
When I look at the data_loader.py,the Dataset_Wiki and Dataset_Solar class should use self.data instead of data so that fit the original code 
original:
self.data = mms.fit_transform(self.data)

fixed:
if type == '1':
            mms = MinMaxScaler(feature_range=(0, 1))
            training_end = int(len(self.data) * self.train_ratio)
            mms.fit(self.data[:training_end])
            self.data = mms.transform(self.data)
